src/umpire/util/memory_sanitizers.hpp

```diff
@@ -0,0 +1,30 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_memory_sanitizers_HPP
+#define UMPIRE_memory_sanitizers_HPP
+
+#if defined(__clang__)
```

Comment:
Don't we want to make this optionally be compiled in?  I thought I saw something about a 73% slowdown?

Reply:
The ASAN macros themselves are no-ops when compiled without `-fsanitize=address`

Reply:
I will add an option to guard it also, since it adds a conditional for the `Platform` check.